service/internal/proctelemetry/config.go

```diff
@@ -0,0 +1,244 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proctelemetry // import "go.opentelemetry.io/collector/service/internal/proctelemetry"
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/exporters/stdout/stdoutmetric"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/aggregation"
+	"go.opentelemetry.io/otel/sdk/resource"
+
+	"go.opentelemetry.io/collector/obsreport"
+	semconv "go.opentelemetry.io/collector/semconv/v1.18.0"
+	"go.opentelemetry.io/collector/service/telemetry"
+)
+
+const (
+	// supported exporters
+	consoleExporter      = "console"
+	otlpExporter         = "otlp"
+	protocolProtobufHTTP = "http/protobuf"
+	protocolProtobufGRPC = "grpc/protobuf"
+	compressionGzip      = "gzip"
+
+	// supported metric readers
+	PrometheusMetricReader = "prometheus"
+	PeriodMetricReader     = "periodic"
+
+	// gRPC Instrumentation Name
+	GRPCInstrumentation = "go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
+
+	// http Instrumentation Name
+	HTTPInstrumentation = "go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+var (
+	// GRPCUnacceptableKeyValues is a list of high cardinality grpc attributes that should be filtered out.
+	GRPCUnacceptableKeyValues = []attribute.KeyValue{
+		attribute.String(semconv.AttributeNetSockPeerAddr, ""),
+		attribute.String(semconv.AttributeNetSockPeerPort, ""),
+		attribute.String(semconv.AttributeNetSockPeerName, ""),
+	}
+
+	// HTTPUnacceptableKeyValues is a list of high cardinality http attributes that should be filtered out.
+	HTTPUnacceptableKeyValues = []attribute.KeyValue{
+		attribute.String(semconv.AttributeNetHostName, ""),
+		attribute.String(semconv.AttributeNetHostPort, ""),
+	}
+)
+
+func toStringMap(in map[string]interface{}) map[string]string {
+	out := map[string]string{}
+	for k, v := range in {
+		out[k] = fmt.Sprintf("%v", v)
+	}
+	return out
+}
+
+func newOTLPGRPCMetricExporter(ctx context.Context, args map[string]interface{}) (sdkmetric.Exporter, error) {
+	opts := []otlpmetricgrpc.Option{}
+	for k, v := range args {
+		switch k {
```

Comment:
IDK what the configuration file spec looks like, but if a user configures an unsupported value should we error?

Reply:
i think it would make sense for an error to be returned for unsupported values